util/environment/environment_test.go

```diff
@@ -99,3 +99,34 @@ func TestMergeEnvironments(t *testing.T) {
 		})
 	}
 }
+
+func TestKey(t *testing.T) {
+	tests := []struct {
+		name string
+		s    string
+		want string
+	}{
+		{
+			name: "special characters are replaced",
+			s:    "FOO@BAR",
+			want: "FOO_BAR",
```

Comment:
I think that if we're strictly following POSIX, `@` would be allowed in a variable name since it's a portable character.
See https://stackoverflow.com/a/2821183/5086091, the second quote in particular.

I don't think it's an issue that needs to be fixed right now, but strict POSIX compatibility would mean we allow way more "special" characters.

Reply:
Yeah, I was a bit confused by all the mixed information and decided to take the safe approach and only allow alphanumeric characters and `_`. That being said, we can always decide later to allow `@` if it causes any problem.